src/logic/wallets/tests/ethAddresses.test.ts

```diff
@@ -0,0 +1,281 @@
+    // given
+    const userAddress = 'address1'
+    const owners = List([makeOwner({ address: userAddress })])
+    const safeInstance = makeSafe({ owners })
+    const expectedResult = true
+
+    // when
+    const result = isUserAnOwner(safeInstance, userAddress)
+
+    // then
+    expect(result).toBe(expectedResult)
+  })
+  it('Given a safe with 1 owner, and an userAddress different to that owner, returns true ', () => {
+    // given
+    const userAddress = 'address1'
+    const userAddress2 = 'address2'
+    const owners = List([makeOwner({ address: userAddress })])
+    const safeInstance = makeSafe({ owners })
+    const expectedResult = false
+
+    // when
+    const result = isUserAnOwner(safeInstance, userAddress2)
+
+    // then
+    expect(result).toBe(expectedResult)
+  })
+})
+
+describe('isUserAnOwnerOfAnySafe', () => {
+  it('Given a list of safes, one of them has an owner equal to the userAddress, returns true', () => {
+    // given
+    const userAddress = 'address1'
+    const userAddress2 = 'address2'
+    const owners1 = List([makeOwner({ address: userAddress })])
+    const owners2 = List([makeOwner({ address: userAddress2 })])
+    const safeInstance = makeSafe({ owners: owners1 })
+    const safeInstance2 = makeSafe({ owners: owners2 })
+    const safesList = List([safeInstance, safeInstance2])
+    const expectedResult = true
+
+    // when
+    const result = isUserAnOwnerOfAnySafe(safesList, userAddress)
+
+    // then
+    expect(result).toBe(expectedResult)
+  })
+  it('Given a list of safes, none of them has an owner equal to the userAddress, returns false', () => {
+    // given
+    const userAddress = 'address1'
+    const userAddress2 = 'address2'
+    const userAddress3 = 'address3'
+    const owners1 = List([makeOwner({ address: userAddress3 })])
+    const owners2 = List([makeOwner({ address: userAddress2 })])
+    const safeInstance = makeSafe({ owners: owners1 })
+    const safeInstance2 = makeSafe({ owners: owners2 })
+    const safesList = List([safeInstance, safeInstance2])
+    const expectedResult = false
+
+    // when
+    const result = isUserAnOwnerOfAnySafe(safesList, userAddress)
+
+    // then
+    expect(result).toBe(expectedResult)
+  })
+})
+
+describe('isValidEnsName', () => {
+  it('Given no ens name, returns false', () => {
+    // given
+    const ensName = null
+    const expectedResult = false
+
+    // when
+    const result = isValidEnsName(ensName)
+
+    // then
+    expect(result).toBe(expectedResult)
+  })
+  it('Given a ens name not the format [value].[eth|test|xyz|luxe], returns false', () => {
+    // given
+    const ensName = 'test'
+    const expectedResult = false
+
+    // when
+    const result = isValidEnsName(ensName)
+
+    // then
+    expect(result).toBe(expectedResult)
+  })
+  it('Given a ens name not the format [value].[eth|test|xyz|luxe], returns false', () => {
```

Comment:
for this and the previous one, you can be specific in the description of the test.

Say: "Given `test.et12312`, returns false"

Or: "Should return false for `test.et12312`"

Reply:
I would leave the implementation detail for the test and keep the description more descriptive

Reply:
so you're Ok with two (or more) test cases with the same description?

Or you're proposing a different wording? If so, can you please exemplify it?

Reply:
First test case:
Given an ens name missing a domain name returns false

Second test case:
Given an ens name with an invalid domain returns false

Reply:
To add more options, I usually like the it -> should wording. (That is why the function is called like that)

First test case:
it('should return false for an ens without extension')

Second test case:
it('should return false for an ens with an invalid extension')

so if you read the whole semantic would say:

isValidEnsName:
it should return false for an ens without extension,
it should return false for an ens with an invalid extension'